juniper_official/chassis/check-fan-state-rpm.rule

```diff
@@ -35,24 +35,52 @@ healthbot {
              */			
             field fan-name {
                 sensor fan-netconf {
-                    path Item;
+                    path name;
                 }
                 type string;
                 description "Name of the fan tray and fan number";
             }
-            field measurement {
+            field measurements {
                 sensor fan-netconf {
                     path measurement;
+                    zero-suppression;
+                    data-if-missing {
+                        value 0;
+                    }					
+                }
+                type string;
+                description "RPM String Value";				
+            }			
+            field measurement {
+                formula {
+                    capture-groups {
+                        field-name "$measurements";
+                        pattern "\d+";
+                    }
                 }
                 type integer;
-                description "RPM measurement";
+                description "RPM Value";				
             }
-            field rpm-percent {
+            field rpm-percents {
```

Comment:
please retain same name

Reply:
rpm-percent field is still present and uses capture-group formula to derive integer value from rpm-percents.
rpm-percents is new field which is a string type.